test/unit/support/azure/service_test.rb

```diff
@@ -0,0 +1,50 @@
+require_relative '../../test_helper'
+require_relative '../../../../libraries/support/azure/service'
+
+describe TestCredentials do
+  before do
+    @service = Object.new
+    @service.extend(Azure::Service)
+  end
+
+  it 'converts simple hashes to structs' do
+    struct = @service.structify({ name: 'John', age: 20 })
+
+    assert_equal('John', struct.name)
+    assert_equal(20, struct.age)
+  end
+
+  it 'converts arrays to structs' do
+    struct = @service.structify([{ name: 'John', age: 20 },
+                                 { name: 'Jane', age: 25 }])
```

Comment:
Not going to fight about it, but I find extracting the setup to `let` statements with explaining variables helps with overall readability both by coercing you to give the different data distinguishing names, but also by removing their setup from the comparisons.